### PR TITLE
Update Dart sample to latest Dart server template

### PR DESCRIPTION
**Pull request by mit-mit**
Update all Dart sample files to follow the latest Dart server template (`dart create -t server-shelf`).

### CODE REVIEW

**Comment by mit-mit** [comment]
@jonasfj can you please review?

File: code-samples/community/serving/helloworld-dart/pubspec.yaml [hunk 1]
```diff
@@ -2,7 +2,14 @@ name: hello_world_dart
 publish_to: none # let's not accidentally publish this to pub.dartlang.org
 
 environment:
-  sdk: '>=2.12.0 <3.0.0'
+  sdk: ">=2.18.0-165.1.beta <3.0.0"
```
Comment (kevmoo) [inline code-samples/community/serving/helloworld-dart/pubspec.yaml:5]: Could we just leave this at `2.17.0`?

Comment (mit-mit) [reply code-samples/community/serving/helloworld-dart/pubspec.yaml]: Yes, good catch!

**Comment by mit-mit** [comment]
@jonasfj is on vacation; @kevmoo can you LGTM?

File: code-samples/community/serving/helloworld-dart/README.md [hunk 2]
```diff
@@ -17,67 +17,24 @@ that you can use for testing. It reads in the env variable `TARGET` and prints
...
-   name: hello_world_dart
-   publish_to: none # let's not accidentally publish this to pub.dartlang.org
-
-   environment:
-     sdk: ">=2.12.0 <3.0.0"
-
-   dependencies:
-     shelf: ^1.0.0
+   ```shell
+   > create -t server-shelf helloworld-dart
```
Comment (hrishikesh-kadam) [inline code-samples/community/serving/helloworld-dart/README.md:27]: Hi @mit-mit!
dart word seems to be missing in the command.

I came to this PR by traversing the following path - 

1. https://www.cloudskillsboost.google/course_sessions/1402704/labs/187125
   The hyperlink at the end of Task 2

2. https://cloud.google.com/run/docs/quickstarts/build-and-deploy
   Write the sample application -> Dart hyperlink

So, someone from the Cloud Team would need to update the above hyperlink once this goes live.

**Review by kevmoo — APPROVED** [review]

File: code-samples/community/serving/helloworld-dart/README.md (hunk 2 above)
Comment (mit-mit) [reply code-samples/community/serving/helloworld-dart/README.md]: > dart word seems to be missing in the command.

oh, good catch, fixed now

**Comment by abrennan89** [comment]
@mit-mit any errors reported will need to be fixed before this can merge: https://github.com/knative/docs/runs/7770884752?check_suite_focus=true

**Comment by mit-mit** [comment]
@abrennan89 I fixed the test issue

**Pull request merged**